packages/hawtio/src/plugins/camel/routes/Source.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { jolokiaService } from '@hawtiosrc/plugins'
+import { CodeEditor, Language } from '@patternfly/react-code-editor'
+import { isRoutesFolder } from '@hawtiosrc/plugins/camel/camel-content-service'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import { MBeanNode } from '@hawtiosrc/plugins/shared'
+
+export const findParentContext = (child: MBeanNode, root: MBeanNode): MBeanNode | undefined => {
+  let found: MBeanNode | undefined
+  root.children?.every(mb => {
+    if (mb.id === child.id) {
+      found = root
+      return false
+    } else found = findParentContext(child, mb)
+    return true
+  })
+
+  return found
+}
+export const Source: React.FunctionComponent = () => {
+  const { selectedNode, tree } = useContext(CamelContext)
+  const [xmlSource, setxmlSource] = useState<string>('')
+
+  useEffect(() => {
+    if (selectedNode) {
+      if (isRoutesFolder(selectedNode)) {
```

Comment:
Is this necessary given that the Source View should not even be displayed if the selectedNode is not a routes folder?

Reply:
that's true, It's just that selected node has `MBean | null` type so isRouteFolder method signature would need to be changed then.

Reply:
No. `isRouteFolder` is able to stay as that signature if this `useEffect` has an `if (!selectedNode) return null` clause at the beginning, eg. [here](https://github.com/hawtio/hawtio-next/blob/main/packages/hawtio/src/plugins/camel/exchanges/InflightExchanges.tsx#L15).

Reply:
Ah so I misread your comment before. The think is that Source component is displayed for both route folders and routes itself. That's why I have it there.

Reply:
Indeed. I see your point. What we could do is if `xml` is cached as a property on the `routes` node in the same manner as the `else` clause of this file (L34 below) then we will not have to differentiate between the types of node. Feel free to ping me if that makes no sense.